common/lexer.c

```diff
@@ -121,7 +121,7 @@ p11_lexer_next (p11_lexer *lexer,
 				lexer->tok_type = TOK_PEM;
 				lexer->tok.pem.begin = lexer->at;
 				lexer->tok.pem.length = end - lexer->at;
-				assert (end - lexer->at <= lexer->remaining);
+				assert (end - lexer->at <= (int)lexer->remaining);
```

Comment:
perhaps `ptrdiff_t` might be more appropriate? not sure if it is available in MSVC though.